html/components/spinners.js

```diff
@@ -27,12 +28,24 @@ const getSpinnerClasses = (options) => {
 const setSpinning = (element, options) => {
   const el = element;
   const width = element.offsetWidth;
+
   const spinningAriaLabel = options.ariaLabel || 'Loading';
-  el.setAttribute('data-sprk-spinner-text', el.textContent);
+  const ariaValueText = options.ariaValueText || 'Loading';
+  const role = options.role || 'progressbar';
+
+  el.classList.add('sprk-c-Button--has-spinner');
   el.setAttribute('aria-label', spinningAriaLabel);
-  el.innerHTML = `<div class="${getSpinnerClasses(options)}"></div>`;
+  el.setAttribute('data-sprk-spinner-text', el.textContent);
+  el.setAttribute('disabled', 'true');
```

Comment:
Hey @RVMendoza , for the disabled attr here, it only needs to be present on the element and doesn't need to be set to "true". You can change 'true' to an empty string `setAttribute('disabled', '');`. More info here if you like: https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute